### PR TITLE
add transport parameters description

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 1]
```diff
@@ -142,6 +142,19 @@ transport = AioHttpTransport(
...
+
+| Parameters | Description |
+| --- | --- |
+| connection_timeout | A single float in seconds for the connection timeout. Defaults to 300 seconds. |
+| read_timeout | A single float in seconds for the read timeout. Defaults to 300 seconds. |
+| connection_verify | SSL certificate verification. Enabled by default. Set to False to disable, alternatively can be set to the path to a CA_BUNDLE file or directory with certificates of trusted CAs. |
+| connection_cert | Client-side certificates. You can specify a local cert to use as client side certificate, as a single file (containing the private key and the certificate) or as a tuple of both files' paths. |
+| proxies | Dictionary mapping protocol or protocol and hostname to the URL of the proxy. |
+| cookies | Dict or CookieJar object to send with the `Request`. |
+| stream | whether to immediately download the response content. Defaults to ``False``. |
```
Comment (lmazuel) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:155]: I want to make the difference between what is intended for customer, between what is intended for SDK owner. Steam for example, should not be available to customers but SDK owners only.

If I give this table to a customer, this is misleading.

Also, some of them can be set of the operation level too (like verify I think), we should have the disctinction of that

Comment (xiangyan99) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:155]: I thought about this too. But given the name of the doc is CLIENT_LIBRARY_DEVELOPER.md. I am leaning towards the target audience of this doc is SDK developer and we will let SDK developers decide which information they want to expose to their users. Your ideas?

Comment (lmazuel) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:155]: Fair enough, does it mean we have the public one in the main Readme? What about splitting still into public/private to bring clarity to the SDK writer what will be visible by people.

Comment (xiangyan99) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:155]: My intention is like policies. We put all available ones in the doc for SDK developers. And then we will work with Kraig to work out one for SDK users.

**Pull request merged**